packages/core/core/src/BundlerRunner.js

```diff
@@ -61,11 +61,16 @@ export default class BundlerRunner {
       bundles.map(bundle => this.nameBundle(namers, bundle, bundleGraph))
     );
 
-    let bundlePaths = bundles.map(b => b.filePath);
+    let bundleNames = bundles.map(
+      b =>
+        nullthrows(b.target).name +
+        ':' +
+        normalizeSeparators(nullthrows(b.name))
+    );
     assert.deepEqual(
-      bundlePaths,
-      unique(bundlePaths),
-      'Bundles must have unique filePaths'
```

Comment:
Actually, I think filePath might be a better sanity check. Since it's possible to generate a name with slashes, and distDirs are independent, it's possible that more than one {target,name} combination could pick the same file path. I don't think the converse is possible.

Or maybe we do both? @devongovett @padmaia what do you think?